ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -189,72 +262,304 @@ def waypoints_cb(self, lane_msg):
+
+            if curpt.get_s() <= JMT_instance.final_displacement:
+                # create the main part of the jmt curve
+                t = self.gen_point_in_jmt_curve(curpt, jmt_ptr, t)
+                if self.check_point(curpt) is True:
+                    recalc = True
+            else:
+                # the car has reached the point where the JMT
+                # curve should reach target velocity of 0.0
+                # we smooth the transition
+                rospy.loginfo("{} beyond S = {} at ptr_id = {}".
+                              format(curpt.get_s(), JMT_instance.
+                                     final_displacement, mod_ptr))
+                self.set_transition_to_stop(mod_ptr)
+        return recalc
+
+    def maintain_speed(self, start_ptr, num_wps):
+        # sets the speed of the car to default velocity for the
+        # range of waypoints
+        for ptr in range(start_ptr, start_ptr + num_wps):
+            mod_ptr = ptr % len(self.waypoints)
+            curpt = self.waypoints[mod_ptr]
+
+            rospy.loginfo("setting to default velocity {} at ptr = {}"
+                          .format(self.default_velocity, mod_ptr))
+
+            velocity = min(curpt.get_maxV(), self.default_velocity)
+            curpt.set_v(velocity)
+            curpt.JMTD.set_VAJt(velocity, 0.0, 0.0, 0.0)
+            curpt.JMT_ptr = -1
+
+    def init_acceleration(self, start_ptr, num_wps):
+        # starts the car moving with a lead foot
+        offset = 0
+
+        velocity = max(0.4, self.waypoints[start_ptr].get_v())
+        self.waypoints[start_ptr].set_v(velocity)
+        self.waypoints[start_ptr].JMTD.set_VAJt(
+                    velocity, self.default_accel, 0.0, 0.0)
+        offset += 1
+        while velocity < 1.5 and offset < num_wps:
+            disp = self.waypoints[start_ptr + offset].get_s() -\
+                self.waypoints[start_ptr].get_s()
+            velocity = min(max(0.5, math.sqrt(self.default_accel * disp * 2.0)),
+                           self.waypoints[start_ptr + offset].get_maxV())
+            rospy.loginfo("velocity set to {} using accel = {} and "
+                          "disp = {} at ptr = {}"
+                          .format(velocity, self.default_accel, disp,
+                                  start_ptr + offset))
+            self.waypoints[start_ptr + offset].set_v(velocity)
+            self.waypoints[start_ptr + offset].JMTD.set_VAJt(
+                velocity, self.default_accel, 0.0, 0.0)
+            self.waypoints[start_ptr + offset].JMT_ptr = -1
+            offset += 1
+        return offset - 1
+
+    def generate_speedup(self, start_ptr, end_ptr):
+        # this is triggered after initiate_aceleration()
+        # gets the car moving - that way we can adjust them
+        # independently
+        recalc = False
+        curpt = self.waypoints[start_ptr]
+
+        if curpt.JMT_ptr == -1 or self.state != 'speedup':
```

Comment:
When the vehicle is still within the initial accel waypoints, the curpt.JMT_ptr is set to -1 so that the JMT speedup profile is continuously regenerated and keeps incrementing to another JMT_ptr until the car finally gets to the first JMT point where curpt.JMT_ptr is set to the real JMT ID value.  Either this condition should change to account for being in the initial accel waypoint zone, or the JMT_ptr of the initial accel waypoints could be set to the JMT_ptr ID of the JMT part of the profile to prevent this.

Reply:
I had already put the fix in earlier today (but not pushed, waiting for this) - I just check to see if state != 'speedup' before setting JMT_ptr to -1 in init_accel - so it only happens once.